evm

```diff
@@ -118,23 +118,79 @@ function evm_system() {
   export PATH
 }
 
-function _unzip-strip() {
+function _unzip_strip() {
   # shopt is a bash built-in command. Zsh, has setopt instead
   # so, set the command according to the type of shell (only supports bash and zsh)
   local SHOPT="shopt"
-  if [[ "$EVM_PS_SHELL" == *zsh ]]; then
+  if [[ "$EVM_PS_SHELL" == *zsh ]] || [[  "$(which $SHELL)" == *"/zsh" ]] ; then
```

Comment:
Why do you need the second test ? When would `$EVM_PS_SHELL` be set with something different that the second test is required?

Reply:
I noticed the the variable $EVM_PS_SHELL is actually null or empty, has nothing, doing which SHELL seems like it did the trick. So checking   [[ "" == *zsh ]] is false

Reply:
That seems odd, because the code defining that variable:

```
if [[ -z "${SHELL}" ]]; then
  EVM_PS_SHELL="$(ps | grep $$)"
else
  EVM_PS_SHELL="$SHELL"
fi
```

Will only create an empty value if the `SHELL` variable is empty and `ps | grep $$` also returns nothing. And your solution also uses the `SHELL` variable to figure out the current shell. It does not make much sense to me that the original code would create an empty variable and  then `which $SHELL`  would solve it.

Reply:
could be, could be that is not needed at all. Feel free to remove those too